File: knock.js
import { createServer } from 'net';

const setupFlagServer = () => {
    const server = createServer((connection) => {
        connection.on('data', (data) => {
            const message = data.toString().trim();
            if (message === 'KNOCK_KNOCK') {
                connection.write('CTF{p0rt_kn0ck3r_m4st3r}\n');
            } else {
                connection.write('Try again!\n');
            }
            connection.end(); // Close the connection after responding
        });
    });

    server.listen(31337, '0.0.0.0', () => {
        console.log('Server listening on port 31337');
    });
};

setupFlagServer();

File: knock.py
import socket

def setup_flag_server():
    server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    server.bind(('0.0.0.0', 31337))
    server.listen(1)
    
    while True:
        connection, address = server.accept()
        data = connection.recv(1024).decode()
        
        if data.strip() == "KNOCK_KNOCK":
            connection.send(b"CTF{p0rt_kn0ck3r_m4st3r}\n")
        else:
            connection.send(b"Try again!\n")
        
        connection.close()

if __name__ == "__main__":
    setup_flag_server()